src/Sulu/Bundle/AdminBundle/Admin/Navigation/NavigationItem.php

```diff
@@ -11,6 +11,9 @@
 
 namespace Sulu\Bundle\AdminBundle\Admin\Navigation;
 
+/**
+ * @implements \Iterator<int|null, self>
```

Comment:
I think there is no case a iterator can has a `null` as key so this would be:

```suggestion
 * @implements \Iterator<int, self>
```

Reply:
If you don't set the position and just call $this->key(), then it will return null. Maybe initalizing it to 0 would make more sense? The we don't need the null in the type definition of `position`.

Reply:
Got point! I think it would make sense to set `position = 0;` because `var_dump([null => 'hello']);` is converted to `['' => 'hello']` which is neither null or int. https://3v4l.org/WImgU